File: exercism-solutions/clojure/collatz-conjecture/src/collatz_conjecture.clj
(ns collatz-conjecture)

(defn collatz [num]
  (if (<= num 0)
    (throw (AssertionError. "number needs to be positive"))
    ((fn [x n]
       (if (= x 1) n
           (recur
            (if (even? x)
              (/ x 2)
              (+ (* x 3) 1))
            (inc n))))
     num 0)))

File: exercism-solutions/julia/collatz-conjecture/collatz-conjecture.jl

function collatz_steps(n)
    if n < 1
        throw(DomainError("number needs to be >= 1"))
    end

    i = 0
    while n > 1
        n = iseven(n) ? div(n, 2) : 3n + 1
        i += 1
    end
    return i
end
